controllers/configmap_controller.go

```diff
@@ -477,6 +482,98 @@ func (r *ConfigMapReconciler) EnsureServicesConfigMapExists() error {
 	return r.createServicesConfigMapOnBootup()
 }
 
+// EnsureWICDRBAC ensures the WICD RBAC resources exist as expected, creates them if not
+func (r *ConfigMapReconciler) EnsureWICDRBAC() error {
```

Comment:
I suggest you have a function `ensureWICDRoleBinding()` that you call from here. Only if that succeeds will you call `ensureWICDClusterRoleBinding()`. This will make the flow much easier to follow.

Reply:
Makes sense, updated